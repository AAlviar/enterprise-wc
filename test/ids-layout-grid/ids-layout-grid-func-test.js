/**
 * @jest-environment jsdom
 */
import IdsLayoutGrid from '../../src/ids-layout-grid/ids-layout-grid';
import IdsLayoutColumn from '../../src/ids-layout-grid/ids-layout-column';

describe('IdsLayoutGrid Component', () => {
  let gridElem;

  beforeEach(async () => {
    const grid = new IdsLayoutGrid();
    const column1 = new IdsLayoutColumn();
    const column2 = new IdsLayoutColumn();
    const column3 = new IdsLayoutColumn();

    document.body.appendChild(grid);

    grid.appendChild(column1);
    grid.appendChild(column2);
    grid.appendChild(column3);

    gridElem = document.querySelector('ids-layout-grid');
  });

  afterEach(async () => {
    document.body.innerHTML = '';
  });

  it('renders with no errors', () => {
    const errors = jest.spyOn(global.console, 'error');
    let elem = new IdsLayoutGrid();
    document.body.appendChild(elem);
    elem.remove();

    elem = new IdsLayoutColumn();
    document.body.appendChild(elem);
    elem.remove();
    expect(document.querySelectorAll('ids-layout-grid').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    expect(gridElem.outerHTML).toMatchSnapshot();
    gridElem.fixed = true;
    expect(gridElem.outerHTML).toMatchSnapshot();
  });

  it('renders fixed setting', () => {
    const elem = new IdsLayoutGrid();
    elem.fixed = true;
    document.body.appendChild(elem);
    expect(elem.fixed).toEqual('true');
    expect(document.querySelectorAll('.ids-fixed').length).toEqual(1);
  });

  it('renders fill setting', () => {
    const col = new IdsLayoutColumn();
    col.fill = true;
    document.body.appendChild(col);
    expect(col.fill).toEqual('true');
    expect(document.querySelectorAll('.ids-background-fill').length).toEqual(1);
  });

  it('renders fixed setting then removes it', () => {
    const elem = new IdsLayoutGrid();
    elem.fixed = true;
    document.body.appendChild(elem);
    elem.fixed = false;
    expect(elem.fixed).toEqual(null);
    expect(document.querySelectorAll('.ids-fixed').length).toEqual(0);
  });

  it('renders fill setting then removes it', () => {
    const col = new IdsLayoutColumn();
    document.body.appendChild(col);
    col.fill = true;
    col.fill = false;
    expect(col.fill).toEqual(null);
    expect(document.querySelectorAll('.ids-background-fill').length).toEqual(0);
  });

  it('renders auto setting', () => {
    const elem = new IdsLayoutGrid();
    elem.auto = true;
    document.body.appendChild(elem);
    expect(elem.auto).toEqual('true');
    expect(document.querySelectorAll('.ids-layout-cols-auto').length).toEqual(1);
  });

  it('renders auto setting then removes it', () => {
    const elem = new IdsLayoutGrid();
    elem.auto = true;
    elem.auto = false;
    document.body.appendChild(elem);
    expect(elem.auto).toEqual(null);
    expect(document.querySelectorAll('.ids-layout-cols-auto').length).toEqual(0);
  });

  it('renders cols setting', () => {
    const elem = new IdsLayoutGrid();
    elem.cols = 16;
    document.body.appendChild(elem);
    expect(elem.cols).toEqual('16');
    expect(document.querySelectorAll('.ids-layout-cols-16').length).toEqual(1);
  });

  it('renders rows setting', () => {
    const elem = new IdsLayoutGrid();
    elem.rows = 4;
    document.body.appendChild(elem);
    expect(elem.rows).toEqual('4');
    expect(document.querySelectorAll('.ids-layout-rows-4').length).toEqual(1);
  });

  it('renders col-span setting', () => {
    const col = new IdsLayoutColumn();
    col.colSpan = 4;
    document.body.appendChild(col);
    expect(col.colSpan).toEqual('4');
    expect(document.querySelectorAll('.ids-layout-col-span-4').length).toEqual(1);
  });

  it('renders col-start setting', () => {
    const col = new IdsLayoutColumn();
    col.colStart = 4;
    document.body.appendChild(col);
    expect(col.colStart).toEqual('4');
    expect(document.querySelectorAll('.ids-layout-col-start-4').length).toEqual(1);
  });

  it('renders col-end setting', () => {
    const col = new IdsLayoutColumn();
    col.colEnd = 4;
    document.body.appendChild(col);
    expect(col.colEnd).toEqual('4');
    expect(document.querySelectorAll('.ids-layout-col-end-4').length).toEqual(1);
  });

  it('renders row-span setting', () => {
    const col = new IdsLayoutColumn();
    col.rowSpan = 4;
    document.body.appendChild(col);
    expect(col.rowSpan).toEqual('4');
    expect(document.querySelectorAll('.ids-layout-row-span-4').length).toEqual(1);
  });

  it('renders row-start setting', () => {
    const col = new IdsLayoutColumn();
    col.rowStart = 4;
    document.body.appendChild(col);
    expect(col.rowStart).toEqual('4');
    expect(document.querySelectorAll('.ids-layout-row-start-4').length).toEqual(1);
  });

  it('renders row-end setting', () => {
    const col = new IdsLayoutColumn();
    col.rowEnd = 4;
    document.body.appendChild(col);
    expect(col.rowEnd).toEqual('4');
    expect(document.querySelectorAll('.ids-layout-row-end-4').length).toEqual(1);
  });
});
