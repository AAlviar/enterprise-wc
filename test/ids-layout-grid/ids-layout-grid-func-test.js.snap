// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsLayoutGrid Component renders correctly 1`] = `"<ids-layout-grid class=\\"ids-layout-grid\\"><style id=\\"ids-layout-grid-styles\\">[object Object]</style><ids-layout-column class=\\"ids-layout-column\\"></ids-layout-column><ids-layout-column class=\\"ids-layout-column\\"></ids-layout-column><ids-layout-column class=\\"ids-layout-column\\"></ids-layout-column></ids-layout-grid>"`;

exports[`IdsLayoutGrid Component renders correctly 2`] = `"<ids-layout-grid class=\\"ids-layout-grid ids-fixed\\" fixed=\\"true\\"><style id=\\"ids-layout-grid-styles\\">[object Object]</style><ids-layout-column class=\\"ids-layout-column\\"></ids-layout-column><ids-layout-column class=\\"ids-layout-column\\"></ids-layout-column><ids-layout-column class=\\"ids-layout-column\\"></ids-layout-column></ids-layout-grid>"`;
